bw2analyzer/lci.py

```diff
@@ -18,17 +64,63 @@ def get_labeled_inventory(lca: LCA) -> pd.DataFrame:
         lca, "inventory"
     ), "Must calculate life cycle inventory first. Please call lci()."
 
-    rows = [
-        get_activity(lca.dicts.biosphere.reversed[i]).as_dict()
-        for i in range(lca.inventory.shape[0])
-    ]
-    columns = [
-        get_activity(lca.dicts.activity.reversed[i]).as_dict()
-        for i in range(lca.inventory.shape[1])
-    ]
-
-    return pd.DataFrame(
-        data=lca.inventory.todense(),
-        index=pd.MultiIndex.from_frame(pd.DataFrame(rows)),
-        columns=pd.MultiIndex.from_frame(pd.DataFrame(columns)),
-    )
+    # get activity metadata
+    meta = _load_database_metadata(lca, usecols)
+
+    if wide_format:
+
+        # map local matrix indices to global database ids
+        rows_global = lca.dicts.biosphere.reversed.values()
+        cols_global = lca.dicts.activity.reversed.values()
+
+        # associate metadata, drop columns which are empty
+        row_meta = (
+            pd.DataFrame(index=rows_global, data=rows_global, columns=["id"])
+            .join(meta, how="left")
+            .dropna(how="all", axis=1)
+        )
+        col_meta = (
+            pd.DataFrame(index=cols_global, data=cols_global, columns=["id"])
+            .join(meta, how="left")
+            .dropna(how="all", axis=1)
+        )
+
+        # make dataframe with appropriate row and column indices
+        df = pd.DataFrame(
+            data=lca.inventory.todense(),
+            index=pd.MultiIndex.from_frame(row_meta),
+            columns=pd.MultiIndex.from_frame(col_meta),
+        )
+        return df
+    # return dataframe in long format
+    else:
+        rows, cols = lca.inventory.nonzero()
+
+        # map local matrix indices to global database ids
+        rows_global = [lca.dicts.biosphere.reversed[r] for r in rows]
+        cols_global = [lca.dicts.activity.reversed[c] for c in cols]
+
+        # filter metadata
+        row_meta = (
+            pd.DataFrame(index=rows_global, data=rows_global, columns=["id"])
+            .join(meta, how="left")
+            .reset_index(drop=True)
+        )
+        col_meta = (
+            pd.DataFrame(index=cols_global, data=cols_global, columns=["id"])
+            .join(meta, how="left")
+            .reset_index(drop=True)
+        )
+
+        # rename columns to avoid duplicate names for row and col
```

Comment:
Can you briefly describe why you think this is preferred versus adding another level to the multiindex?

Reply:
Not sure, what exactly you are referring to. The `id` is already a separate column. Here, I am dropping the index to be able to concatenate row and column metadata.

Reply:
Sorry, maybe I am just off in my own world, but can't multiindices be hierarchical? I would find it much cleaner to have ('foo', 'name') than 'foo_name'. But maybe pandas doesn't actually work like this. I have to admit I am having flashbacks to when others have done these types of string manipulations and I had to reverse engineer them, it wasn't fun...

Reply:
Ah, you are referring to the MultiIndex names. Well, no the names can't be hierarchical. It might be possible to assign tuples as names but then standard methods like `query` won't work. I would strongly advice against it.